include/wlr/types/wlr_output_management_v1.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_OUTPUT_MANAGEMENT_V1_H
+#define WLR_TYPES_WLR_OUTPUT_MANAGEMENT_V1_H
+
+#include <stdbool.h>
+#include <wayland-server.h>
+#include <wlr/types/wlr_output.h>
+
+struct wlr_output_manager_v1 {
+	struct wl_display *display;
+	struct wl_global *global;
+	struct wl_list resources; // wl_resource_get_link
+
+	struct wl_list heads; // wlr_output_head_v1::link
+	uint32_t serial;
+
+	struct {
+		struct wl_signal apply; // wlr_output_configuration_v1
+		struct wl_signal test; // wlr_output_configuration_v1
+		struct wl_signal destroy;
+	} events;
+
+	struct wl_listener display_destroy;
+
+	void *data;
+};
+
+struct wlr_output_head_v1_state {
+	struct wlr_output *output;
+
+	bool enabled;
+	struct wlr_output_mode *mode;
+	struct {
+		int width, height;
+		int refresh;
+	} custom_mode;
+	int32_t x, y;
+	enum wl_output_transform transform;
+	double scale;
+};
+
+struct wlr_output_head_v1 {
+	struct wlr_output_head_v1_state state;
+	struct wlr_output_manager_v1 *manager;
+	struct wl_list link; // wlr_output_manager_v1::heads
+
+	struct wl_list resources; // wl_resource_get_link
+	struct wl_list mode_resources; // wl_resource_get_link
+
+	struct wl_listener output_destroy;
+};
+
+struct wlr_output_configuration_v1 {
+	struct wl_list heads; // wlr_output_configuration_head_v1::link
+
+	// client state
+	struct wlr_output_manager_v1 *manager;
+	uint32_t serial;
+	bool finalized; // client has requested to apply the config
+	bool finished; // feedback has been sent by the compositor
+	struct wl_resource *resource; // can be NULL if destroyed early
+};
+
+struct wlr_output_configuration_head_v1 {
+	struct wlr_output_head_v1_state state;
+	struct wlr_output_configuration_v1 *config;
+	struct wl_list link; // wlr_output_configuration_v1::heads
+
+	// client state
+	struct wl_resource *resource; // can be NULL if finalized or disabled
+
+	struct wl_listener output_destroy;
+};
+
+struct wlr_output_manager_v1 *wlr_output_manager_v1_create(
+	struct wl_display *display);
+void wlr_output_manager_v1_set_configuration(
```

Comment:
I think there should be a comment here, saying that the `config` parameter and its heads might be freed. This certainly isn't obvious, except if you read the code of the function.

Reply:
Ohh right. I really wanted to document this and forgot. Fixed!